drivers/volume/portworx/portworx.go

```diff
@@ -625,6 +627,81 @@ func (d *portworx) WaitForUpgrade(n node.Node, image, tag string) error {
 	return nil
 }
 
+func (d *portworx) GetReplicationFactor(vol *torpedovolume.Volume) (int64, error) {
+	name := d.schedOps.GetVolumeName(vol)
+	t := func() (interface{}, bool, error) {
+		vols, err := d.volDriver.Inspect([]string{name})
+		if err != nil && err == volume.ErrEnoEnt {
+			return 0, false, volume.ErrEnoEnt
+		} else if err != nil {
+			return 0, false, err
+		}
+		if len(vols) == 1 {
+			return vols[0].Spec.HaLevel, false, nil
+		}
+		return 0, false, fmt.Errorf("Extra volumes with the same volume name/ID seen") //Shouldn't reach this line
+	}
+
+	tmp, err := task.DoRetryWithTimeout(t, defaultRetryInterval, defaultRetryInterval)
```

Comment:
change the second arg to use some retry timeout, instead of interval
nit: can we not call this tmp? may be call iRepFactor to indicate it is an interface